### ISSUE TITLE
Add track filtering to the tc_rmw tool.

### ISSUE THREAD

**Issue by JohnHalleyGotway** [dtcenter/MET#1315]
## Describe the Enhancement ##
While developing a METplus wrapper for the tc_rmw tool, several issues with the existing tc_rmw tool were identified, that should be corrected:
1. The "data_file_list" command line option does not work as expected.
2. Removed unused variables from tc_rmw.h, including references to bdecks.
3. Add an update history header block to tc_rmw.cc.
4. Remove unused reference to "adeck.nc".
5. TC-RMW should filter the input track data down to a single track.  Currently the "model" option from the config file is not actually parsed or used.  Use that, as well as other track data filtering options (e.g. storm_id, basin, cyclone, and init_time), to filter the input track data down to a single track.
  - 0 tracks should result in an error message.
  - \>1 tracks should result in a warning message indicating that the first track will be used.

### Time Estimate ###
1 day.

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
No sub-issues required.

### Relevant Deadlines ###
None.

### Funding Source ###
Charge 2780601 by Friday 5/1.

### Assignee ###
- [X] Select **engineer(s)** or **no engineer** required (John HG)
- [X] Select **scientist(s)** or **no scientist** required (Tracy)

### Labels ###
- [X] Select **component(s)**
- [X] Select **priority**
- [X] Select **requestor(s)**

### Projects and Milestone ###
- [X] Review **projects** and select relevant **Repository** and **Organization** ones
- [X] Select **milestone**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [X] [METplus](https://github.com/NCAR/METplus/issues/new/choose), [MET](https://github.com/NCAR/MET/issues/new/choose), [METdb](https://github.com/NCAR/METdb/issues/new/choose), [METviewer](https://github.com/NCAR/METviewer/issues/new/choose), [METexpress](https://github.com/NCAR/METexpress/issues/new/choose), [METcalcpy](https://github.com/NCAR/METcalcpy/issues/new/choose), [METplotpy](https://github.com/NCAR/METplotpy/issues/new/choose)
No impacts to other repos.

## Enhancement Checklist ##
See the [METplus Workflow](https://ncar.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [X] Complete the issue definition above.
- [x] Fork this repository or create a branch of **develop**
Branch name:  `feature_<Issue Number>_<Description>`: feature_1315_tcrmw
- [x] Complete the development and test your changes.
- [x] Add/update unit tests.
- [x] Add/update documentation.
- [x] Push local changes to GitHub.
- [x] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [x] Iterate until the reviewer(s) accept and merge your changes.
- [x] Delete your fork or branch.
- [x] Close this issue.

**Comment by JohnHalleyGotway** [comment]
Development on 05/15/20:
- Added header block.
- Providing a data file list is now working.
- Removed unused BDECK vars.
- Replaced nm_to_km with existing vx_math constant.
- Removed "adeck.nc" reference.
- Renamed the "-adeck" command line option to "-deck", but still quietly support "-adeck".
- Added lots of filtering options and updated the code to parse and apply them.
- Still need to:
  - Update the documentation.
  - Test this new functionality.
  - Update the unit tests to make exercise the track filters with both ADECK and BDECK tracks.

**Comment by JohnHalleyGotway** [comment]
I submitted a PR for this work: https://github.com/NCAR/MET/pull/1360
After the PR was approved, I merged into the develop branch.
But those changes caused the NB to fail because of some real differences in the tc_rmw output. Specifically, the track lines are now missing from the NetCDF output. Need to fix this and submit a new PR. The NB will still fail because of the addition of the _FillValue variable attribute and a new unit test run. But those changes are expected and desired.
